pkgs/by-name/de/deno/package.nix

```
@@ -46,12 +50,19 @@ rustPlatform.buildRustPackage rec {
     installShellFiles
   ];
 
-  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [
-    apple-sdk_11
-    # V8 supports 10.15+; binary references `aligned_alloc` directly
-    (darwinMinVersionHook "10.15")
+  configureFlags = lib.optionals stdenv.cc.isClang [
+    # This never worked with clang, but became a hard error recently: https://github.com/llvm/llvm-project/commit/3d5b610c864c8f5980eaa16c22b71ff1cf462fae
+    "--disable-multi-os-directory"
   ];
 
+  buildInputs =
+    [ libffi ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      apple-sdk_11
+      # V8 supports 10.15+; binary references `aligned_alloc` directly
+      (darwinMinVersionHook "10.15")
+    ];
```

```suggestion
  buildInputs = [ libffi ];
```

both x64 and aarch64 are now using the same sdk (11.3) so this is no longer needed. -- oh, but I was probably just the formatter that made these lines looked changed...

---

I can remove the SDK package, but I'm not sure about the min version hook (I'm not sure I fully understand the implications of the deployment target setting).

I looked it up in the stdenv docs, and it says the following:

> On macOS, the `darwinMinVersion` and `darwinSdkVersion` are always the same, and are currently set to 11.3.

So removing the hook would _increase_ the minimum supported version to 11.3, right?

---

it can be removed. it was probably just added for x64 darwin to bump it the least amount to get things to compile. but everything is at 11 now so I can't see the point in lowering it.

---

Done, I think it's ready for merge now